Mathlib/LinearAlgebra/Projection.lean

```diff
@@ -424,6 +425,41 @@ theorem eq_conj_prod_map' {f : E →ₗ[R] E} (h : IsProj p f) :
   · simp only [coe_prodEquivOfIsCompl, comp_apply, coe_inr, coprod_apply, map_zero,
       coe_subtype, zero_add, map_coe_ker, prodMap_apply, zero_apply, add_zero]
 
+theorem isIdempotentElem {R V : Type*} [Semiring R] [AddCommMonoid V] [Module R V]
```

Comment:
For this theorem and those below, I think you can make use of the existing variables `S` and `M`.